src/storage/blockchain/onchain-issuer-adapter/non-merklized/version/v0.0.1/onchain-non-merklized-issuer-adapter.ts

```diff
@@ -0,0 +1,309 @@
+        return isSupported ? null : interfaceObj.name;
+      })
+    );
+
+    const unsupportedInterfacesFiltered = unsupportedInterfaces.filter(
+      (interfaceName) => interfaceName !== null
+    );
+
+    if (unsupportedInterfacesFiltered.length > 0) {
+      throw new Error(`Unsupported interfaces: ${unsupportedInterfacesFiltered.join(', ')}`);
+    }
+  }
+
+  /**
+   * Retrieves a credential from the on-chain non-merklized contract.
+   * @param userId The user's core.Id.
+   * @param credentialId The unique identifier of the credential.
+   */
+  public async getCredential(
+    userId: Id,
+    credentialId: bigint
+  ): Promise<{
+    credentialData: INonMerklizedIssuer.CredentialDataStructOutput;
+    coreClaimBigInts: bigint[];
+    credentialSubjectFields: INonMerklizedIssuer.SubjectFieldStructOutput[];
+  }> {
+    const [credentialData, coreClaimBigInts, credentialSubjectFields] =
+      await this._contract.getCredential(userId.bigInt(), credentialId);
+    return { credentialData, coreClaimBigInts, credentialSubjectFields };
+  }
+
+  /**
+   * Converts on-chain credential to a verifiable credential.
+   *
+   * @param credentialData Data structure of the credential from the contract.
+   * @param coreClaimBigInts Claim data in bigint format.
+   * @param credentialSubjectFields Subject fields of the credential.
+   */
+  public async convertOnChainInfoToW3CCredential(
+    credentialData: INonMerklizedIssuer.CredentialDataStructOutput,
+    coreClaimBigInts: bigint[],
+    credentialSubjectFields: INonMerklizedIssuer.SubjectFieldStructOutput[]
+  ): Promise<W3CCredential> {
+    const c = new Claim().unMarshalJson(JSON.stringify(coreClaimBigInts.map((b) => b.toString())));
+
+    const credentialSubject = await this.convertCredentialSubject(
+      c,
+      credentialData.context,
+      credentialData._type,
+      credentialSubjectFields
+    );
+
+    const credentialRequest = {
+      id: this.credentialId(credentialData.id),
+      credentialSchema: credentialData.credentialSchema.id,
+      type: credentialData._type,
+      credentialSubject: credentialSubject,
+      expiration: c.getExpirationDate()?.getTime(),
+      displayMethod: this.convertDisplayMethod(credentialData.displayMethod),
+      context: credentialData.context,
+      revocationOpts: {
+        id: this._contractAddress,
+        nonce: Number(c.getRevocationNonce()),
+        type: CredentialStatusType.Iden3OnchainSparseMerkleTreeProof2023
+      },
+      issuanceDate: getDateFromUnixTimestamp(Number(credentialData.issuanceDate)).getTime()
+    };
+
+    const existenceProof = await this.existenceProof(c);
+    const w3c = new W3CCredential().fromCredentialRequest(this._issuerDid, credentialRequest);
+    w3c.proof = [existenceProof];
+    return w3c;
+  }
+
+  private credentialId(id: bigint): string {
+    return `urn:iden3:onchain:${this._chainId}:${this._contractAddress}:${id}`;
```

Comment:
is urn:iden3:onchain:  agreed somewhere?

Reply:
I'm not sure. I can't find a place where we made a commitment to this format. But Golang lib uses the same format. I think we agreed this format before.